### DISCUSSION TITLE
Define datatype for make_geocube

### DISCUSSION THREAD

**Discussion by mjyshin** [corteva/geocube#61] [Ideas]
Hello, I have a lot of Shapely polygons as `shapes` that I want to rasterize into an output image, but the output will be huge. Is there a way to use `make_geocube` so that when I run something like
```
gdf = gpd.GeoDataFrame({'mask': shape_types}, geometry=shapes)
cube = make_geocube(gdf, resolution=(-1, 1), fill=0)
mask = cube.mask.values
```
the `mask` will be an int8 array instead of float64 (`shape_types` will naturally only be small integers)? Thanks in advance!

**Comment by snowman2** [comment]
You can convert it afterwards:
```python
mask = cube.mask.values.astype("int8")
```

**Reply by mjyshin** [reply]
Thank you for the reply! I was hoping to bypass some memory issues; is there no way to specify it beforehand to ensure Numpy won't try to initialize a giant ndarray of floats?

**Reply by snowman2** [reply]
It is currently defaulted to `float64` [ref](https://github.com/corteva/geocube/blob/2c07ea1bb2503228aaea2cd9a562d81483e96771/geocube/rasterize.py#L85). If you would like to specify a different dtype, you could open up a PR to add the `dtype` kwarg to `rasterize_image` and then use `functools.partial` to change the dtype similar to what is done in this example [ref](https://corteva.github.io/geocube/stable/examples/rasterize_point_data.html).

**Reply by snowman2** [reply]
See: #82 and #98
